app/models/report.rb

```diff
@@ -15,14 +16,55 @@ def self.generate_report_for_keywords(search_keywords)
     )
   end
 
+  def self.find_for_guid(guid)
+    where(guid: guid).first
+  end
+
+  def find_occupation(occupation_id)
+    occupations.find { |o| o.id == occupation_id }
+  end
+
   def mark_occupations_as_selected(occupation_ids)
     occupations.where(id: occupation_ids).select do |occupation|
       occupation.selected = true
       occupation.save
     end
+    occupations.where.not(id: occupation_ids).select do |occupation|
+      occupation.selected = false
+      occupation.accepted = false
+      occupation.save
+    end
+  end
+
+  def occupations_to_review
+    occupations.to_a.select(&:selected)
+      .uniq(&:soc_occupation_id)
+      .select { |o| o.accepted.nil? }
```

Comment:
A few comments on this:

Why are we doing a distinct on soc_occupation_id – does this mean that we actually want to be returning SocOccupations here? If not, then perhaps the domain model could be improved, e.g. why is occupation through keywords, if we just ignore duplicate occupations?

Also, regardless of the above, why not use ActiveRecord again here:

If we actually want to return SocOccupations (altohugh this should be moved to a scope on occupations in that case):
```ruby
SocOccupation.joins(:occupations).merges(Occupations.where(selected: true, accepted: nil)).distinct
```

or if we want Occupations (and this can probably be improved to move the uniq into the DB, but I don't know how to do that without playing):
```ruby
occupations.where(selected: true, accepted: nil).include(:soc_occupation).uniq(&:soc_occupation)
```